File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/Payment.java
package org.morecup.jimmerddd.java.sample.domain.order;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.OneToOne;
import org.morecup.jimmerddd.java.sample.domain.base.BaseEntity;
import java.math.BigDecimal;

@Entity
public interface Payment extends BaseEntity {
    BigDecimal amount();
    String paymentMethod();

    @OneToOne
    Order order();
}

File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/Payment.kt
package org.morecup.jimmerddd.kotlin.sample.domain.order

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.OneToOne
import org.morecup.jimmerddd.kotlin.sample.domain.base.BaseEntity
import java.math.BigDecimal

@Entity
interface Payment:BaseEntity {
    val amount: BigDecimal
    val paymentMethod: String

    @OneToOne
    val order: Order
}